From 0e107f7bac8d84d8fb2bb5b83bfb4474348a82c1 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@smile.fr>
Date: Wed, 17 Apr 2019 22:47:05 +0200
Subject: [PATCH] meson: use wayland-scanner provided by wayland-scanner-path

When cross-compiling the .pc file might point to the wrong
wayland-scanner binary (target rather than host) resulting in a
non-executable and wrong scanner.
First try by using the path provided by wayland-scanner-path,
and if that fails fall back into pkg-config.

Based on the patch for autotools provided by Gustavo Zacarias.

Signed-off-by: Romain Naour <romain.naour@smile.fr>
---
 meson.build       | 5 ++++-
 meson_options.txt | 6 ++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6c4f1508855..ef0532b2cfb 100644
--- a/meson.build
+++ b/meson.build
@@ -1319,7 +1319,10 @@ endif
 
 if with_platform_wayland
   dep_wl_scanner = dependency('wayland-scanner', native: true)
-  prog_wl_scanner = find_program(dep_wl_scanner.get_pkgconfig_variable('wayland_scanner'))
+  prog_wl_scanner = get_option('wayland-scanner-path')
+  if prog_wl_scanner == ''
+    prog_wl_scanner = find_program(dep_wl_scanner.get_pkgconfig_variable('wayland_scanner'))
+  endif
   if dep_wl_scanner.version().version_compare('>= 1.15')
     wl_scanner_arg = 'private-code'
   else
diff --git a/meson_options.txt b/meson_options.txt
index ccf70659ff7..59ff8df8337 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -324,3 +324,9 @@ option(
   value : true,
   description : 'Enable direct rendering in GLX and EGL for DRI',
 )
+option(
+  'wayland-scanner-path',
+  type : 'string',
+  value : '',
+  description : 'Locations to wayland-scanner binary. Default: use path from pkg-config'
+)
-- 
2.20.1

